march_hardware/include/march_hardware/ethercat/slave.h

```diff
@@ -31,7 +31,7 @@ class Slave : public PdoSlaveInterface
 
   ~Slave() noexcept override = default;
 
-  uint16_t getSlaveIndex() const
+  int getSlaveIndex() const
```

Comment:
Already fixed ;) 
(originally the parallel of this method in MotorController would return -1 for controllers that are not ethercat slaves, but I refactored the ethercat master so that that is no longer necessary)